napi.h

```diff
@@ -151,28 +171,61 @@ namespace Napi {
     double DoubleValue() const;
   };
 
+  /// A JavaScript string or symbol value (that can be used as a property name).
   class Name : public Value {
   public:
-    Name();
-    Name(napi_env env, napi_value value);
+    Name(); ///< Creates a new _empty_ Name instance.
+    Name(napi_env env, napi_value value); ///< Wraps a N-API value primitive.
   };
 
+  /// A JavaScript string value.
   class String : public Name {
   public:
-    static String New(napi_env env, const std::string& value);
-    static String New(napi_env env, const std::u16string& value);
-    static String New(napi_env env, const char* value);
-    static String New(napi_env env, const char16_t* value);
-    static String New(napi_env env, const char* value, size_t length);
-    static String New(napi_env env, const char16_t* value, size_t length);
-
-    String();
-    String(napi_env env, napi_value value);
-
-    operator std::string() const;
-    operator std::u16string() const;
-    std::string Utf8Value() const;
-    std::u16string Utf16Value() const;
+    /// Creates a new String value from a UTF-8 encoded C++ string.
+    static String New(
+      napi_env env,            ///< N-API environment
+      const std::string& value ///< UTF-8 encoded C++ string
+    );
+
+    /// Creates a new String value from a UTF-16 encoded C++ string.
+    static String New(
+      napi_env env,               ///< N-API environment
+      const std::u16string& value ///< UTF-16 encoded C++ string
+    );
+
+    /// Creates a new String value from a UTF-8 encoded C string.
+    static String New(
+      napi_env env,     ///< N-API environment
+      const char* value ///< UTF-8 encoded C string
```

Comment:
C string seems to imply "null-terminated" but should we explicitly mention that here since there is an overload which takes a length parameter?